client/config/webpack.common.js

```diff
@@ -101,6 +102,17 @@ module.exports = {
       new webpack.DefinePlugin({
         "process.env.NODE_ENV": JSON.stringify(process.env.NODE_ENV)
       }),
+      new CircularDependencyPlugin({
+        // exclude detection of files based on a RegExp
+        exclude: /node_modules/,
+        // add errors to webpack instead of warnings
+        failOnError: true,
+        // allow import cycles that include an asyncronous import,
+        // e.g. via import(/* webpackMode: "weak" */ './file.js')
+        allowAsyncCycles: false,
```

Comment:
this comment `// allow import cycles that include an asyncronous import,` and this value `allowAsyncCycles: false` seems contradicting ?

Reply:
I thought so too; it's a straight copy-paste from https://github.com/aackerman/circular-dependency-plugin#basic-usage though. ;-)

Reply:
Yeah, i checked for further explanations but didn't find any. Although, the library's code seems to be pointing to a contradiction as well. https://github.com/aackerman/circular-dependency-plugin/blob/master/index.js line:103
```
      // ignore dependencies that are resolved asynchronously
      if (this.options.allowAsyncCycles && dependency.weak) { continue }
```

Reply:
Unless you come to a definite conclusion of whether `false` actually means `true`, I'd say we leave it.

Reply:
Huh i got it, it means "Change this field if you want to allow it". That part is little confusing.